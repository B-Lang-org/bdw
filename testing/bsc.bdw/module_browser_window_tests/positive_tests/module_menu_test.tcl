
# Opens the specified project
handlers::open_project project.bspec 

while {$BSPEC(FILES) == ""} {
        update
}

# Compiles the current project
handlers::compile_file Mult2.bsv -withdeps

# Waits until the process with the 'BSPEC(BUILDPID)' id has finished
while {$BSPEC(BUILDPID) != ""} {
        update
}

# Loads the specified module
handlers::load_module mkMult2

# Reloads the currently loaded module
handlers::reload_module

# Cleans the build results
commands::full_clean

# Closes the current project
handlers::close_project

# Exits the workstation
exit
