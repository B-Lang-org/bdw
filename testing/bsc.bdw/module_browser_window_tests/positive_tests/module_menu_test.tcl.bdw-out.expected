Executing: bsc -u -sim -simdir mult4 -bdir mult4 -info-dir . -keep-fires -p .:%/Libraries  mult4/Mult2.bsv
WARNING: + bsc -u -sim -simdir mult4 -bdir mult4 -info-dir . -keep-fires -p .:%/Libraries mult4/Mult2.bsv
RESULT: checking package dependencies
RESULT: compiling mult4/Mult.bsv
RESULT: compiling mult4/Mult2.bsv
RESULT: code generation for mkMult2 starts
WARNING: Warning: "mult4/Mult2.bsv", line 23, column 8: (G0117)
WARNING:   Rule `start' shadows the effects of `cycle' when they execute in the same
WARNING:   clock cycle. Affected method calls:
WARNING:     product.write, mcand.write, mplr.write
WARNING:   To silence this warning, use the `-no-warn-action-shadowing' flag.
RESULT: Elaborated module file created: mult4/mkMult2.ba
RESULT: All packages are up to date.
mkMult2
mkMult2
Mult2 Mult PreludeBSV Prelude
mkMult2
